src/test/java/io/confluent/connect/jdbc/sink/integration/PostgresDatatypeIT.java

```diff
@@ -42,67 +45,163 @@
+        TimeUnit.MINUTES.toMillis(3));
+
+    ConsumerRecords<byte[], byte[]> records = connect.kafka().consume(1, CONSUME_MAX_DURATION_MS,
+        DLQ_TOPIC_NAME);
+
+    assertEquals(1, records.count());
   }
 
   @Test
-  public void testWriteToTableWithUuidColumn() throws SQLException {
+  public void testRecordSchemaMoreFieldsThanTableSendsToErrorReporter() throws Exception {
+    props.put(ERRORS_TOLERANCE_CONFIG, ToleranceType.ALL.value());
+    props.put(DLQ_TOPIC_NAME_CONFIG, DLQ_TOPIC_NAME);
+    props.put(DLQ_TOPIC_REPLICATION_FACTOR_CONFIG, "1");
+
+    createTableWithLessFields();
+    connect.configureConnector("jdbc-sink-connector", props);
+    waitForConnectorToStart("jdbc-sink-connector", 1);
+
+    final Schema schema = SchemaBuilder.struct().name("com.example.Person")
+        .field("firstname", Schema.STRING_SCHEMA)
+        .field("lastname", Schema.STRING_SCHEMA)
+        .field("jsonid", Schema.STRING_SCHEMA)
+        .field("userid", Schema.STRING_SCHEMA)
+        .build();
+    final Struct struct = new Struct(schema)
+        .put("firstname", "Christina")
+        .put("lastname", "Brams")
+        .put("jsonid", "5")
+        .put("userid", UUID.randomUUID().toString());
+
+    String kafkaValue = new String(jsonConverter.fromConnectData(tableName, schema, struct));
+    connect.kafka().produce(tableName, null, kafkaValue);
+
+    waitForCommittedRecords("jdbc-sink-connector", Collections.singleton(tableName), 1, 1,
+        TimeUnit.MINUTES.toMillis(2));
+
+    ConsumerRecords<byte[], byte[]> records = connect.kafka().consume(1, CONSUME_MAX_DURATION_MS,
+        DLQ_TOPIC_NAME);
+
+    assertEquals(1, records.count());
+  }
+
+  @Test
+  public void testWriteToTableWithUuidColumn() throws Exception {
     createTableWithUuidColumns();
-    startTask();
+    connect.configureConnector("jdbc-sink-connector", props);
+    waitForConnectorToStart("jdbc-sink-connector", 1);
+
     final Schema schema = SchemaBuilder.struct().name("com.example.Person")
                                        .field("firstname", Schema.STRING_SCHEMA)
                                        .field("lastname", Schema.STRING_SCHEMA)
                                        .field("jsonid", Schema.STRING_SCHEMA)
                                        .field("userid", Schema.STRING_SCHEMA)
                                        .build();
-    UUID uuid = UUID.randomUUID();
-    String jsonid = "5";
     final Struct struct = new Struct(schema)
         .put("firstname", "Christina")
         .put("lastname", "Brams")
-        .put("jsonid", jsonid)
-        .put("userid", uuid.toString());
-    task.put(Collections.singleton(new SinkRecord(tableName, 1, null, null, schema, struct, 1)));
+        .put("jsonid", "5")
+        .put("userid", UUID.randomUUID().toString());
+
+    String kafkaValue = new String(jsonConverter.fromConnectData(tableName, schema, struct));
+    connect.kafka().produce(tableName, null, kafkaValue);
```

Comment:
Can we replace this with a call to `produceRecord`?

Reply:
Done. I/ve checked the other tests to make sure I've replaced it properly there.